forest_benchmarking/rpe.py

```diff
@@ -1,227 +1,491 @@
+            which is given by the relative phase between the two eigenvector components.
+        4) Invert the change of basis to return to the computational basis.
+        5) Prepare (one of) the qubit(s) for measurement along either the X or Y axis.
+        6) Measure this qubit, and in the multi-qubit case other qubits participating in rotation.
+    Measure_qubits can be used e.g. in the case of noisy cross-talk to measure the effective
+    action of some "rotation program" that acts on completely different qubits but nonetheless
+    rotates each measure_qubit.
+
+    The single qubit algorithm is due to:
 
     [RPE]  Robust Calibration of a Universal Single-Qubit Gate-Set via Robust Phase Estimation
             Kimmel et al., Phys. Rev. A 92, 062315 (2015)
             https://doi.org/10.1103/PhysRevA.92.062315
+            arXiv:1502.02677
 
     [RPE2] Experimental Demonstration of a Cheap and Accurate Phase Estimation
             Rudinger et al., Phys. Rev. Lett. 118, 190502 (2017)
             https://doi.org/10.1103/PhysRevLett.118.190502
-
-    :param rotation: the program or gate whose angle of rotation is to be estimated
-    :param num_depths: the number of depths in the protocol described in [RPE]. Max depth = 2**(num_depths-1)
-    :param axis: the axis of rotation specified by (theta, phi). Assumed to be the Z axis if none is specified.
-    :param measurement_qubit: Pertinent only to CPHASE experiment. See generate_2q_single_depth_experiment above
-    :param init_one: Pertinent only to CPHASE experiment. See generate_2q_single_depth_experiment above
-    :return:
+            arXiv:1702.01763v1
+
+    :param rotation: the program or gate whose angle of rotation is to be estimated. Note that
+        this program will be run through forest_benchmarking.compilation.basic_compile().
+    :param change_of_basis: a matrix, gate, or program for the unitary change of basis
+        transformation which maps the computational basis into the basis formed by eigenvectors
+        of the rotation. The sign of the estimate will be determined by which computational basis
+        states are mapped to which eigenvectors. Following the right-hand-rule convention,
+        a rotation of RX(phi) for phi>0 about the +X axis should be paired with a change of basis
+        maps |0> --> |+> and |1> --> |-> . This is achieved by the gate RY(pi/2, qubit).
+    :param num_depths: the number of depths in the protocol described in [RPE]. A depth is the
+        number of consecutive applications of the rotation in a single iteration. The maximum
+        depth is 2**(num_depths-1)
+    :param measure_qubits: the qubits whose angle of rotation, as a result of the action of
+        the rotation program, RPE will attempt to estimate. These are the only qubits measured.
+    :param prepare_and_post_select: is a bitstring used only in the multi-qubit case where one
```

Comment:
Removing the default all-eigenvector superposition option boils down to simply requiring this argument. Doing so would probably simplify some of the logic in both acquire_data and estimation methods. On the one hand, the all-eigenvectors option allows for one experiment to give all 4 estimates. However, if we know it always performs worse with realistic noise models then removing it as an option could simplify things. E.g. something I haven't really handled well so far is that you should probably multiply the number of shots by the number of eigenvectors in order to match the RPE bounds/procedure.

Reply:
Ok, well add it to a todo list or make an issue :)